smbclient/_pool.py

```diff
@@ -148,10 +148,14 @@ def lookup_domain(self, domain_name):  # type: (str) -> Optional[DomainEntry]
             if domain.domain_name.lower() == ("\\" + domain_name.lower()):
                 return domain
 
+    def clear_expired_cache(self):
```

Comment:
Thanks for working on this, sorry I haven't gotten to it myself. 1 minor nit:

* Can we make this `_clear_expired_cache` as it shouldn't really be something that is used outside of the internal lookups

Reply:
Yea I merged it in because I assumed I was going to push my own PR that renamed it and also did the expiry check for the domain controller lookup but unfortunately I was held up with some other stuff. Feel free to submit another PR, otherwise I'll get to it sometime this week.

> But the other side, maybe it's premature optimization, what do you think?

I think what you have here is fine, it will weed out the expired referrals when a lookup occurs and I would hope getting the expiry time wouldn't be too hard. I would say unless it's a known problem we probably don't need any further optimisations here.